src/app/zap-templates/zcl/data-model/chip/joint-fabric-datastore-cluster.xml

```diff
@@ -0,0 +1,298 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2024 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="DatastoreStateEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="Pending" value="0x00"/>
+    <item name="Committed" value="0x01"/>
+    <item name="DeletePending" value="0x02"/>
+    <item name="CommitFailed" value="0x03"/>
+  </enum>
+
+  <enum name="FailureCodeEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="OK" value="0x00"/>
+  </enum>
+
+  <struct name="DatastoreStatusEntry">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="State" type="DatastoreStateEnum" default="Pending"/>
+    <item fieldId="1" name="UpdateTimestamp" type="epoch_us" isNullable="true"/>
+    <item fieldId="2" name="FailureCode" type="FailureCodeEnum" default="OK"/>
+  </struct>
+
+  <struct name="DatastoreNodeKeyEntry">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="GroupKeySetId" type="int16u"/>
+    <item fieldId="1" name="StatusEntry" type="DatastoreStatusEntry"/>
+  </struct>
+
+  <struct name="DatastoreGroupInformationEntry" isFabricScoped="true">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="GroupID" type="int64u"/>
+    <item fieldId="1" name="FriendlyName" type="char_string" length="32"/>
+    <item fieldId="2" name="GroupKeySetID" type="int16u"/>
+    <item fieldId="3" name="GroupCAT" type="int16u"/>
+    <item fieldId="4" name="GroupCATVersion" type="int16u"/>
+    <item fieldId="5" name="GroupPermission" type="AccessControlEntryPrivilegeEnum"/>
+  </struct>
+
+  <struct name="DatastoreBindingEntry">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="ListID" type="int16u"/>
+    <!-- TODO: Need to support WrappedStructs when dealing with FabricScoped items -->
```

Comment:
the fields are fabric-sensitive as well, does that help?

Reply:
There is a build failure if this argument is uncommented. I believe it's related to a Decode method that is unable to decode wrapped struct elements in a message. We may need to proceed with that line commented out for the time being and later uncomment it once the issue is fixed.